test/state/system_contracts.cpp

```diff
@@ -36,17 +36,19 @@ static_assert(std::ranges::is_sorted(SYSTEM_CONTRACTS,
 
 }  // namespace
 
-void system_call(State& state, const BlockInfo& block, evmc_revision rev, evmc::VM& vm)
+StateDiff system_call(
+    const StateView& state, const BlockInfo& block, evmc_revision rev, evmc::VM& vm)
 {
+    State ss{state};
```

Comment:
These names are because of the prototype quality. My proposal for every place:

```cpp
StateView state_view;
State intra_state;
TestState test_state;
```

Reply:
Actually, not so much of this is left so I'll change just this instance.